<?xml version="1.0"?>
<launch>
    <arg name="file_path" default="$(env HOME)/bagfiles/experimental_file_20220224" />
    <arg name="config_path" default="$(find multi_localizer)/config" />

    <!-- roomba -->
    <arg name="is_roomba1"  default="true" />
    <arg name="is_roomba2"  default="true" />
	<arg name="is_roomba3"  default="true" />
	<arg name="is_roomba4"  default="true" />
	<arg name="is_roomba5"  default="true" />
	<arg name="is_roomba6"  default="true" />

    <!--map_server -->
	<arg name="is_map" default="false" />
	<group if="$(arg is_map)" >
		<node pkg="map_server" type="map_server" name="map" args="$(arg config_path)/map/spare/map.yaml" />
	</group>

    <!-- bag -->
    <arg name="is_bag" default="false" />
    <group if="$(arg is_bag)" >
        <include file="$(find multi_localizer)/launch/utils/20220224.launch" >
            <arg name="is_map"     value="false" />
			<arg name="is_rviz"    value="false" />

			<!-- roomba -->
			<arg name="is_roomba1" value="$(arg is_roomba1)" />
			<arg name="is_roomba2" value="$(arg is_roomba2)" />
			<arg name="is_roomba3" value="$(arg is_roomba3)" />
			<arg name="is_roomba4" value="$(arg is_roomba4)" />
			<arg name="is_roomba5" value="$(arg is_roomba5)" />
			<arg name="is_roomba6" value="$(arg is_roomba6)" />
        </include>
    </group>

    <!-- image_unzipper -->
    <include file="$(find multi_localizer)/launch/utils/image_unzipper.launch" >
        <arg name="topic_name" value="detected_image" />
        <arg name="is_rviz"    value="false" />

        <!-- roomba -->
		<arg name="is_roomba1" value="$(arg is_roomba1)" />
		<arg name="is_roomba2" value="$(arg is_roomba2)" />
		<arg name="is_roomba3" value="$(arg is_roomba3)" />
		<arg name="is_roomba4" value="$(arg is_roomba4)" />
		<arg name="is_roomba5" value="$(arg is_roomba5)" />
		<arg name="is_roomba6" value="$(arg is_roomba6)" />
    </include>

    <node pkg="multi_localizer" type="image_connector" name="image_connector" output="screen" >
        <!-- param -->
        <rosparam command="load" file="$(arg config_path)/params/robot_element.yaml" />
    </node>

    <!-- rviz -->
    <arg name="is_rviz" default="false" />
    <group if="$(arg is_rviz)" >
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg config_path)/rviz/rviz/image_connector.rviz" />
    </group>

</launch>